pkg/sql/catalog/descs/descriptor.go

```diff
@@ -398,139 +518,13 @@ func (tc *Collection) finalizeDescriptors(
 	return nil
 }
 
-// hydrateDescriptors ensures that the descriptors in the slice are hydrated.
-//
-// Callers expect the descriptors to come back hydrated.
-// In practice, array types here are not hydrated, and that's a bummer.
-// Nobody presently is upset about it, but it's not a good thing.
-// Ideally we'd have a clearer contract regarding hydration and the values
-// stored in the various maps inside the collection. One might want to
-// store only hydrated values in the various maps. This turns out to be
-// somewhat tricky because we'd need to make sure to properly re-hydrate
-// all the relevant descriptors when a type descriptor change. Leased
-// descriptors are at least as tricky, plus, there we have a cache that
-// works relatively well.
-//
-// TODO(ajwerner): Sort out the hydration mess; define clearly what is
-// hydrated where and test the API boundary accordingly.
-func (tc *Collection) hydrateDescriptors(
-	ctx context.Context, txn *kv.Txn, flags tree.CommonLookupFlags, descs []catalog.Descriptor,
-) error {
-	for i, desc := range descs {
-		hd, isHydratable := desc.(catalog.HydratableDescriptor)
-		if !isHydratable {
-			continue
-		}
-		var err error
-		descs[i], err = tc.hydrateTypesInDescWithOptions(ctx, txn, hd, flags.IncludeOffline, flags.AvoidLeased)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
```

Comment:
This has all been moved to `hydrate.go` now that hydration takes place exclusively in `getDescriptorsByID`.